sdk/Pulumi/Deployment/Deployment_ReadResource.cs

```diff
@@ -42,7 +42,7 @@ public partial class Deployment
             // Now run the operation, serializing the invocation if necessary.
             var response = await this.Monitor.ReadResourceAsync(resource, request).ConfigureAwait(false);
 
-            return (response.Urn, id, response.Properties, ImmutableDictionary<string, ImmutableHashSet<Resource>>.Empty);
+            return (response.Urn, id, response.Properties, ImmutableDictionary<string, ImmutableHashSet<Resource>>.Empty, Pulumirpc.Result.Success);
```

Comment:
`ReadResource` doesn't have a result field, so we always pretend it's a success.  Not sure if we should pull read and register resource apart here instead of unifying them in a single method.